### PR TITLE
test: Ingester integration tests that can have a little a internal state 

### PR DESCRIPTION
**Pull request by carols10cents**
As a treat.

### CODE REVIEW

File: ingester2/tests/common/mod.rs [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+            .await?
+            .into_inner();
+
+        let record_batch_stream = FlightRecordBatchStream::new_from_flight_data(
+            // convert tonic::Status to FlightError
+            flight_data_stream.map_err(|e| e.into()),
+        );
+
+        let iox_record_batch_stream =
+            influxdb_iox_client::flight::IOxRecordBatchStream::new(record_batch_stream);
```
Comment (domodwyer) [inline ingester2/tests/common/mod.rs:361]: I think the work @alamb did for tidying up the flight client has made this possible/not horrendous 🙏 

I literally couldn't get it working before and gave up.

File: ingester2/tests/write.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        let partition_key = PartitionKey::from("1970-01-01");
+        ctx.write_lp(
+            namespace_name,
+            "bananas greatness=\"unbounded\" 10",
+            partition_key.clone(),
+            0,
+        )
+        .await;
+
+        // A subsequent write with a non-contiguous sequence number to a different table.
```
Comment (domodwyer) [inline ingester2/tests/write.rs:107]: I am looking forward to removing sequence numbers... only to promptly put them back again shortly 😂 😱

File: ingester2/tests/write.rs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        let expected = vec![
+            "+-------+-----------+--------------------------------+",
+            "| count | greatness | time                           |",
+            "+-------+-----------+--------------------------------+",
+            "|       | unbounded | 1970-01-01T00:00:00.000000010Z |",
+            "| 42    |           | 1970-01-01T00:00:00.000000200Z |",
+            "+-------+-----------+--------------------------------+",
+        ];
+        assert_batches_sorted_eq!(&expected, &data);
+    } // Drop the first ingester instance
```
Comment (domodwyer) [inline ingester2/tests/write.rs:146]: Helpful comment - much appreciated :+1:

**Review by domodwyer — APPROVED** [review]
Amazing! Thank you for the effort - this will give us a good framework to drive testing of ingester behaviours across subsystems with a lower noise ratio than e2e and allowing more intrusive testing :+1:

Thank you!

File: ingester2/tests/common/mod.rs (hunk 1 above)
Comment (carols10cents) [reply ingester2/tests/common/mod.rs:361]: Indeed-- thank you so much @alamb !!!!

Comment (alamb) [reply ingester2/tests/common/mod.rs:361]: 👍  Thanks for the shout out -- I am glad it is working more easily ❤️

**Pull request merged**